crates/ruff_workspace/src/configuration.rs

```diff
@@ -395,8 +396,9 @@ impl Configuration {
         })
     }
 
-    pub fn from_options(options: Options, project_root: &Path) -> Result<Self> {
-        warn_about_deprecated_top_level_lint_options(&options.lint_top_level.0);
+    /// Convert the [`Options`] read from the given [`Path`] into a [`Configuration`].
+    pub fn from_options(options: Options, path: &Path, project_root: &Path) -> Result<Self> {
+        warn_about_deprecated_top_level_lint_options(&options.lint_top_level.0, path);
```

Comment:
Do we know why we call this method twice for every configuration ?

Reply:
Presumedly because we call it once at the very start of Ruff to find the "default configuration", and then once when we visited it as part of the directory of files we're analyzing. (We should fix that -- it's not "incorrect" but it is inefficient.)

Reply:
...and we didn't catch this in the tests because we pass --config which bypasses the discovery